src/services/signatureHelp.ts

```diff
@@ -179,457 +179,457 @@ namespace ts.SignatureHelp {
+    }
 
-        function createJavaScriptSignatureHelpItems(argumentInfo: ArgumentListInfo): SignatureHelpItems {
-            if (argumentInfo.invocation.kind !== SyntaxKind.CallExpression) {
-                return undefined;
-            }
+    function createJavaScriptSignatureHelpItems(argumentInfo: ArgumentListInfo, program: Program): SignatureHelpItems {
+        if (argumentInfo.invocation.kind !== SyntaxKind.CallExpression) {
+            return undefined;
+        }
 
-            // See if we can find some symbol with the call expression name that has call signatures.
-            let callExpression = <CallExpression>argumentInfo.invocation;
-            let expression = callExpression.expression;
-            let name = expression.kind === SyntaxKind.Identifier
-                ? <Identifier> expression
-                : expression.kind === SyntaxKind.PropertyAccessExpression
-                    ? (<PropertyAccessExpression>expression).name
-                    : undefined;
+        // See if we can find some symbol with the call expression name that has call signatures.
+        const callExpression = <CallExpression>argumentInfo.invocation;
+        const expression = callExpression.expression;
+        const name = expression.kind === SyntaxKind.Identifier
+            ? <Identifier>expression
+            : expression.kind === SyntaxKind.PropertyAccessExpression
+                ? (<PropertyAccessExpression>expression).name
+                : undefined;
 
-            if (!name || !name.text) {
-                return undefined;
-            }
+        if (!name || !name.text) {
+            return undefined;
+        }
 
-            let typeChecker = program.getTypeChecker();
-            for (let sourceFile of program.getSourceFiles()) {
-                let nameToDeclarations = sourceFile.getNamedDeclarations();
-                let declarations = getProperty(nameToDeclarations, name.text);
-
-                if (declarations) {
-                    for (let declaration of declarations) {
-                        let symbol = declaration.symbol;
-                        if (symbol) {
-                            let type = typeChecker.getTypeOfSymbolAtLocation(symbol, declaration);
-                            if (type) {
-                                let callSignatures = type.getCallSignatures();
-                                if (callSignatures && callSignatures.length) {
-                                    return createSignatureHelpItems(callSignatures, callSignatures[0], argumentInfo);
-                                }
+        const typeChecker = program.getTypeChecker();
+        for (const sourceFile of program.getSourceFiles()) {
+            const nameToDeclarations = sourceFile.getNamedDeclarations();
+            const declarations = getProperty(nameToDeclarations, name.text);
+
+            if (declarations) {

```

Comment:
this is actually code that was there before, i just moved it around. i can get that cleaned in a future PR.